src/main/java/org/jenkinsci/plugins/workflow/multibranch/WorkflowBranchProjectFactory.java

```diff
@@ -30,39 +30,61 @@
 import jenkins.scm.api.SCMProbeStat;
 import jenkins.scm.api.SCMSource;
 import jenkins.scm.api.SCMSourceCriteria;
+import org.apache.commons.lang.StringUtils;
 import org.jenkinsci.plugins.workflow.flow.FlowDefinition;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 
 /**
  * Recognizes and builds {@code Jenkinsfile}.
  */
 public class WorkflowBranchProjectFactory extends AbstractWorkflowBranchProjectFactory {
-    
     static final String SCRIPT = "Jenkinsfile";
+    private String scriptPath = SCRIPT;
 
-    @DataBoundConstructor public WorkflowBranchProjectFactory() {}
+    public Object readResolve() {
+        if (this.scriptPath == null) {
+            this.scriptPath = WorkflowBranchProjectFactory.SCRIPT;
+        }
+        return this;
+    }
+
+    @DataBoundConstructor public WorkflowBranchProjectFactory() { }
+
+    @DataBoundSetter
+    public void setScriptPath(String scriptPath) {
```

Comment:
Nice to have: `FormValidation DescriptorImpl.doCheckScriptPath(@QueryParameter String value)` to complain if you try to, say, use a `\` in the path.